File: mobile/lib/features/deteksi_dini/models/calculator_anjuran_rule.dart
class CalculatorAnjuranRule {
  static const metricBmi = 'bmi';
  static const metricLilaCm = 'lila_cm';
  static const metricYesCount = 'yes_count';
  static const metricZScore = 'z_score';

  static const indicatorHeightForAge = 'height_for_age';
  static const indicatorWeightForAge = 'weight_for_age';
  static const indicatorWeightForHeight = 'weight_for_height';
  static const indicatorPrimary = 'primary';

  static const operatorGt = 'gt';
  static const operatorGte = 'gte';
  static const operatorLt = 'lt';
  static const operatorLte = 'lte';

  CalculatorAnjuranRule({
    required this.sortOrder,
    required this.metric,
    this.indicator,
    this.threshold,
    required this.operator,
    required this.isDefault,
    required this.label,
    this.slug,
    required this.anjuran,
  });

  final int sortOrder;
  final String metric;
  final String? indicator;
  final double? threshold;
  final String operator;
  final bool isDefault;
  final String label;
  final String? slug;
  final String anjuran;

  factory CalculatorAnjuranRule.fromJson(Map<String, dynamic> json) {
    return CalculatorAnjuranRule(
      sortOrder: (json['sort_order'] as num?)?.toInt() ?? 0,
      metric: json['metric'] as String? ?? metricBmi,
      indicator: json['indicator'] as String?,
      threshold: (json['threshold'] as num?)?.toDouble(),
      operator: json['operator'] as String? ?? operatorGt,
      isDefault: json['is_default'] == true,
      label: json['label'] as String? ?? '',
      slug: json['slug'] as String?,
      anjuran: json['anjuran'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'sort_order': sortOrder,
      'metric': metric,
      'indicator': indicator,
      'threshold': threshold,
      'operator': operator,
      'is_default': isDefault,
      'label': label,
      'slug': slug,
      'anjuran': anjuran,
    };
  }
}

class ResolvedAnjuran {
  const ResolvedAnjuran({
    required this.slug,
    required this.label,
    required this.anjuran,
  });

  final String slug;
  final String label;
  final String anjuran;
}

File: backend/app/Models/CalculatorAnjuranRule.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class CalculatorAnjuranRule extends Model
{
	public const METRIC_BMI = 'bmi';

	public const METRIC_LILA_CM = 'lila_cm';

	public const METRIC_YES_COUNT = 'yes_count';

	public const METRIC_Z_SCORE = 'z_score';

	public const INDICATOR_HEIGHT_FOR_AGE = 'height_for_age';

	public const INDICATOR_WEIGHT_FOR_AGE = 'weight_for_age';

	public const INDICATOR_WEIGHT_FOR_HEIGHT = 'weight_for_height';

	public const INDICATOR_PRIMARY = 'primary';

	public const INDICATOR_AGE_10_14 = 'age_10_14';

	public const INDICATOR_AGE_15_17 = 'age_15_17';

	public const INDICATOR_AGE_GT_17 = 'age_gt_17';

	public const OPERATOR_GT = 'gt';

	public const OPERATOR_GTE = 'gte';

	public const OPERATOR_LT = 'lt';

	public const OPERATOR_LTE = 'lte';

	protected $fillable = [
		'education_content_id',
		'sort_order',
		'metric',
		'indicator',
		'threshold',
		'operator',
		'is_default',
		'label',
		'slug',
		'anjuran',
	];

	protected function casts(): array
	{
		return [
			'sort_order' => 'integer',
			'threshold' => 'float',
			'is_default' => 'boolean',
		];
	}

	public function educationContent(): BelongsTo
	{
		return $this->belongsTo(EducationContent::class, 'education_content_id');
	}

	/**
	 * @return array<string, mixed>
	 */
	public function toApiArray(): array
	{
		return [
			'sort_order' => $this->sort_order,
			'metric' => $this->metric,
			'indicator' => $this->indicator,
			'threshold' => $this->threshold,
			'operator' => $this->operator,
			'is_default' => $this->is_default,
			'label' => $this->label,
			'slug' => $this->slug,
			'anjuran' => $this->anjuran,
		];
	}
}
